text/0000-implied-derive.md

```diff
@@ -0,0 +1,425 @@
++ `#[derive(Eq)]` => `#[derive(PartialEq)]`
+
+### Lints
+
++ If you explicitly derive a trait that can be implicitly derived,
+  then a warning named along the lines of `explicitly_derived_super_trait`
+  will be issued recommending that you remove the implictly derived trait.
+  You are encouraged as a user to heed this warning.
+
+### Errors
+
++ `#[derive(Clone, Clone)]` will no longer issue an error about conflicting
+  implementations.
+
+### Custom derive
+
++ This RFC does **not** affect the behavior of `#[derive(..)]` for traits
+  which are custom derived. However, for a more consistent experience across
+  the ecosystem, custom derive macro authors are encouraged to implement
+  super-traits to the extent possible.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Changes
+
+The transitive super-trait closures of all derivable standard library traits are:
+
+```
+Copy => {Clone}
+Clone => {}
+Ord => {Eq, PartialEq, PartialOrd}
+PartialOrd => {PartialEq}
+Eq => {PartialEq}
+PartialEq => {}
+Hash => {}
+Default => {}
+Debug => {}
+```
+
+For each of these traits `T` on the LHS, and all future derivable traits,
+`#[derive(T)]` will now imply `#[derive(C)]` for all `C` in the transitive
+super-trait closure of `T`.
+
+### Union, not concatenation
+
+When the compiler is dealing with the union of all explicitly derived and
+implicitly derived traits, the union of those traits will be derived instead of
+the concatenation. This is particularly important for two reasons:
+
+1. To minimize the breakage in transitioning to edition 2018.
+   If you in the future do `#[derive(Copy, Clone)]`,
+   then the deriving system will see this as `#[derive(Copy, Clone)]`
+   and not `#[derive(Copy, Clone, Clone)]`.
+
+2. `Eq` and `PartialOrd` both imply `PartialEq`, and so if the concatenation is
+   used rather than the union, then the compiler will derive `PartialEq` twice
+   which will result in overlapping impls.
+
+### `#[structural_match]`
+
+[RFC 1445]: https://github.com/rust-lang/rfcs/pull/1445
+
+This RFC supercedes [RFC 1445] in that `#[derive(Eq)]`, where transitively
+applied to the transitive closure of all field types of a type, implies
+`#[structural_match]`. In other words, the requirement of transitively having
+`#[derive(PartialEq, Eq)]` is reduced to `#[derive(Eq)]`.
+
+### Interaction with `#[derive_no_bound]` et al.
+
+[RFC 2353]: https://github.com/rust-lang/rfcs/pull/2353
+
+To ensure consistency, `#[derive_no_bound(Trait)]` as described in [RFC 2353]
+should also imply `#[derive_no_bound(Super)]` where `Super` is the
+transitive super-trait closure of `Trait`. Conversely,
+`#[derive_field_bound(Trait)]` will should imply `#[derive_field_bound(Super)]`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There are two main drawbacks of this proposal.
+
+## Breakage
+
+This RFC will break some code. Specifically, if a user has already manually
```

Comment:
I do not think that this should be considered acceptable. You've used `Clone` as your example, in which case, sure -- That's not realistic breakage that is going to happen. However, it is perfectly reasonable to have a manual impl of `PartialEq`, which is represents full equality, and just want to `#[derive(Eq)]`. I do not think it is worth breaking the stability guarantees that Rust has promised because it leads to other inconveniences.

Reply:
>However, it is perfectly reasonable to have a manual impl of PartialEq, which is represents full equality, and just want to #[derive(Eq)].

That's exactly what I wanted to say as well.
Since `Eq` is just a marker, deriving it is common while manual implementation of `PartialEq` is provided.

Reply:
I think this is a perfectly reasonable objection. However, I think that the scenario you mention will be unlikely, and easily mitigated with just invoking `rustfix` and then it becomes `#[derive(only(Eq))]` which is also more clear wrt. intent. On balance, I think the net win is significant enough to warrant the change.

Reply:
I really don't like seeing `rustfix` mentioned as an argument for breaking changes. a) It's alpha, we cannot rely on it being ready and working perfectly well with the edition, b) it still introduces aggrevation, and rustfix is only a stopgap.

Reply:
The idea of rustfix is to allow people to deal with breakage in an easier way. It is not meant to allow or support breakage in the first place. Rustfix or not, breakage is still bad and should be avoided if possible. Here it is definitely possible and the arguments brought in favour of breakage don't really seem convincing to me.

Reply:
Additionally, while `rustfix` might be able to take over the "change code" phase of a migration, in many if not most cases you will still want or need to review all changes. This becomes more work the more things `rustfix` fixes. And at some point there will be situations where you'll want to move a 10 year old code base to current edition, and you'll have to work through the accumulated changes in 3 editions.